### PR TITLE
wait for cronjob

### PR DESCRIPTION
**Pull request by szaimen**
address https://github.com/nextcloud/all-in-one/pull/2854#discussion_r1265616468

### CODE REVIEW

File: Containers/nextcloud/cron.sh [hunk 1]
```diff
@@ -5,3 +5,6 @@ while true; do
     php -f /var/www/html/cron.php &
     sleep 5m
 done
+
+# Wait for last command to end
+wait $!
```
Comment (szaimen) [inline Containers/nextcloud/cron.sh:10]: @Zoey2936 I have no clue if this actually works...

File: php/containers.json [hunk 2]
```diff
@@ -201,6 +201,7 @@
         "TALK_RECORDING_HOST=nextcloud-aio-talk-recording"
       ],
       "restart": "unless-stopped",
+      "stop_grace_period": 600,
```
Comment (szaimen) [inline php/containers.json:204]: @Zoey2936 do you think 10min are enought to wait for cronjobs to stop?

File: Containers/nextcloud/cron.sh (hunk 1 above)
Comment (Zoey2936) [reply Containers/nextcloud/cron.sh:10]: https://stackoverflow.com/a/3004814 I meant that

Comment (szaimen) [reply Containers/nextcloud/cron.sh:10]: I see and where would I add this inside this script?

Comment (Zoey2936) [reply Containers/nextcloud/cron.sh:10]: see like I did it here in the last 2 lines https://github.com/ZoeyVid/nginx-proxy-manager/blob/develop/rootfs/bin/launch.sh and this: https://stackoverflow.com/a/68931259

Comment (Zoey2936) [reply Containers/nextcloud/cron.sh:10]: not sure if it makes a difference to use 
command1 &
command2 &
wait

or

command1 & 
command2
wait

File: php/containers.json (hunk 2 above)
Comment (szaimen) [reply php/containers.json:204]: @Zoey2936 if wait only waits for the sleep command, 5min should be enough?

File: Containers/nextcloud/cron.sh [hunk 3]
```diff
@@ -4,4 +4,5 @@ set -eu
 while true; do
     php -f /var/www/html/cron.php &
     sleep 5m
+    wait
```
[the remark this answers is not in the document]
Comment (szaimen) [reply Containers/nextcloud/cron.sh:7]: I see but we want the next php command to start already even if the old one is still running. Is that possible?

File: php/containers.json (hunk 2 above)
Comment (Zoey2936) [reply php/containers.json:204]: no

File: Containers/nextcloud/cron.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (Zoey2936) [reply Containers/nextcloud/cron.sh:7]: does the cron php support thst?

File: php/containers.json (hunk 2 above)
Comment (szaimen) [reply php/containers.json:204]: https://docs.docker.com/compose/compose-file/compose-file-v3/#stop_grace_period

File: Containers/nextcloud/cron.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (Zoey2936) [reply Containers/nextcloud/cron.sh:7]: i will check the behaivior later again

File: php/containers.json (hunk 2 above)
Comment (Zoey2936) [reply php/containers.json:204]: and why is this needed here?

File: Containers/nextcloud/cron.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (szaimen) [reply Containers/nextcloud/cron.sh:7]: Thanks!

[the remark this answers is not in the document]
Comment (Zoey2936) [reply Containers/nextcloud/cron.sh:7]: So it seems the script continues first if the second (sleep 5m) finished and it ignores if the first command has finished or not in the current implementation

[the remark this answers is not in the document]
Comment (Zoey2936) [reply Containers/nextcloud/cron.sh:7]: so this PR can be closed

[the remark this answers is not in the document]
Comment (szaimen) [reply Containers/nextcloud/cron.sh:7]: > so this PR can be closed

All right

**Pull request closed without merging**